app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsScreen.kt

```diff
@@ -0,0 +1,152 @@
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImage
+
+private val detailsMapper: MovieDetailsMapper = MovieDetailsMapperImpl()
+
+val movieDetailsViewState = detailsMapper.toMovieDetailsViewState(MoviesMock.getMovieDetails())
+
+@Composable
+fun MovieDetailsRoute(
+) {
+    val movieDetailsViewState by remember { mutableStateOf(movieDetailsViewState) }
+    MovieDetailsScreen(
+        movieDetailsViewState = movieDetailsViewState
+    )
+}
+
+@Composable
+fun MovieDetailsScreen(
+    modifier: Modifier=Modifier,
+    movieDetailsViewState: MovieDetailsViewState,
+) {
+    Column() {
+        LazyColumn(
+            modifier = modifier.fillMaxSize(),
+        ) {
+            item {
+                Box(modifier = Modifier.fillMaxWidth()) {
+                    AsyncImage(
+                        model = movieDetailsViewState.imageUrl,
+                        contentDescription = null,
+                        modifier = Modifier.height(350.dp),
+                        contentScale = ContentScale.FillWidth,
+                    )
+                    Text(
+                        modifier = Modifier
+                            .padding(bottom = 20.dp)
+                            .background(Color.Gray.copy(0.7f), shape = RoundedCornerShape(10.dp))
+                            .padding(start = 3.dp, end = 3.dp)
+                            .align(Alignment.BottomCenter),
+                        text = movieDetailsViewState.title,
+                        fontSize = 20.sp,
+                    )
+                    UserScoreProgressBar(
+                        score = movieDetailsViewState.voteAverage,
+                        modifier = Modifier
+                            .padding(10.dp)
+                            .background(Color.Gray.copy(0.7f), shape = CircleShape)
+                            .size(50.dp)
+                            .align(Alignment.BottomEnd),
+                    )
+                    FavoriteButton(modifier = Modifier
+                        .padding(10.dp)
+                        .size(50.dp)
+                        .align(Alignment.BottomStart),
+                        isFavorite = movieDetailsViewState.isFavorite, onFavoriteButtonClick = {})
```

Comment:
format FavoriteButton
- modifier in new line
- empty line before closing parentheses